scripts/globals/spells/black/meteor.lua

```diff
@@ -14,6 +14,12 @@ spellObject.onMagicCastingCheck = function(caster, target, spell)
     -- TODO: Correct message is "Incorrect job, job level too low, or required ability not activated."  Unable to locate this in our basic or system message functions.
     -- The client blocks the spell via menus, but it can still be cast via text commands, so we have to block it here, albiet with the wrong message.
     if caster:isMob() then
+        -- Promathia has significant fast cast thus need to increase cast time
+        -- to match the retail cast time of 10 seconds
+        if caster:getName() == "Promathia_2" then
+            spell:setCastTime(18000)
+        end
+
```

Comment:
Can we set the cast time in onSpellPrecast instead of putting it in the spell script? Something like this: https://github.com/AirSkyBoat/AirSkyBoat/blob/staging/scripts/zones/Apollyon/mobs/Kaiser_Behemoth.lua#L9

Reply:
> Can we set the cast time in onSpellPrecast instead of putting it in the spell script? Something like this: https://github.com/AirSkyBoat/AirSkyBoat/blob/staging/scripts/zones/Apollyon/mobs/Kaiser_Behemoth.lua#L9

From my understanding onSpellPrecast actually occurs after the casting time and only just before the spell goes off

Reply:
Oh that's too bad! I see that now looking at the invocation site for it. I wonder if it'd be possible to move that to be an actual precast. We would have to review all of the existing use cases. From a quick look it shouldn't be much of a problem but definitely outside the scope of this PR. Maybe we can make a note of it or something?

Reply:
> Oh that's too bad! I see that now looking at the invocation site for it. I wonder if it'd be possible to move that to be an actual precast. We would have to review all of the existing use cases. From a quick look it shouldn't be much of a problem but definitely outside the scope of this PR. Maybe we can make a note of it or something?

Yeah, I do not know much about the use case of that either, could add a separate function as well eventually if needed.